ivy_models/mlpmixer/mlpmixer.py

```diff
@@ -0,0 +1,181 @@
+        self.linear1 = ivy.Linear(mlp_dim, inp_dim)
+        self.dropout = ivy.Dropout(0.25)
+        super(MLPBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.linear0(x)
+        y = self.gelu(y)
+        # y = self.dropout(y)
+        y = self.linear1(y)
+        # y = self.dropout(y)
+        return y
+
+
+class MixerBlock(BaseModel):
+    def __init__(self, hidden_dim, num_patches, token_mlp_dim, channel_mlp_dim):
+        self.norm1 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block1 = MLPBlock(num_patches, token_mlp_dim)
+        self.norm2 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block2 = MLPBlock(hidden_dim, channel_mlp_dim)
+        super(MixerBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.norm1(x)
+        y = ivy.matrix_transpose(y)
+        y = self.mlp_block1(y)
+        y = ivy.matrix_transpose(y)
+        x = x + y
+        y = self.norm2(x)
+        y = x + self.mlp_block2(y)
+        return y
+
+
+class MLPMixer(BaseModel):
+    def __init__(
+        self,
+        vol=72,
+        inp_dim=3,
+        num_classes=10,
+        num_blocks=2,
+        patch_size=16,
+        hidden_dim=128,
+        token_mlp_dim=64,
+        channel_mlp_dim=32,
+        data_format="NCHW",
+        spec=None,
+        v=None,
+    ):
+        self.spec = (
+            spec
+            if spec and isinstance(spec, MLPMixerSpec)
+            else MLPMixerSpec(
+                vol=vol,
+                inp_dim=inp_dim,
+                num_classes=num_classes,
+                num_blocks=num_blocks,
+                patch_size=patch_size,
+                hidden_dim=hidden_dim,
+                token_mlp_dim=token_mlp_dim,
+                channel_mlp_dim=channel_mlp_dim,
+                data_format=data_format,
+            )
+        )
+        self.num_blocks = num_blocks
+        self.inp_dim = inp_dim
+        self.hidden_dim = hidden_dim
+        self.patch_size = patch_size
+        self.vol = vol
+        self.token_mlp_dim = token_mlp_dim
+        self.channel_mlp_dim = channel_mlp_dim
+        self.num_classes = num_classes
+
+        super(MLPMixer, self).__init__(v=v)
+
+    def _build(self, *args, **kwargs):
+        self.conv = ivy.Conv2D(
+            self.inp_dim,
+            self.hidden_dim,
+            [self.patch_size, self.patch_size],
+            [self.patch_size, self.patch_size],
+            0,
+            data_format="NCHW",
+        )
+        self.num_patches = int((self.vol / self.patch_size) ** 2)
+        self.mixer_block = MixerBlock(
+            self.hidden_dim, self.num_patches, self.token_mlp_dim, self.channel_mlp_dim
+        )
+        self.norm = ivy.LayerNorm([self.hidden_dim])
+        self.dropout = ivy.Dropout(0.25)
+        self.linear = ivy.Linear(self.hidden_dim, self.num_classes)
+
+    @classmethod
+    def get_spec_class(self):
+        return MLPMixerSpec
+
+    def _forward(self, x):
+        x = self.conv(x)
```

Comment:
(please double check but) as we would like to support dual format, I think there should be an additional check here, something like, 
```
def _forward(self, x, data_format=None):
        data_format = data_format if data_format else self.spec.data_format
        if data_format == "NHWC":
            x = ivy.permute_dims(x, (0, 3, 1, 2))
        .....
```
please ref alexnet model for more details!